quinn/src/platform/unix.rs

```diff
@@ -317,3 +315,10 @@ fn decode_recv(
         ecn: EcnCodepoint::from_bits(ecn_bits),
     }
 }
+
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+// Chosen somewhat arbitrarily; might benefit from additional tuning.
+pub const BATCH_SIZE: usize = 32;
```

Comment:
So I think this means we're allocating a 2 MB buffer by default, right? I feel like 8 or 16 would be good enough pending actual data.

Reply:
IIRC I did dial this in somewhat for sending, though my benchmarking setup makes things a bit too noisy to make this easy. Instead, I've adjusted the max UDP payload size to something realistic for the internet, which ends up with us allocating less memory than before despite the batching.

Reply:
Keep in mind that this is for a full endpoint, which can handle N connections. 2MB for 1000 connections (on a server) would e.g. be not a lot, but could even be undersized.  For an applicaton it's a different story - but since it needs the 2MB only once and it's still not a lot compared to socket send and receive buffers it won't matter.

I downloaded the branch, and gathered some stats on large the vectors are. Definitely seen a lot of receives which carried around 30 datagrams, so it isn't unused.

Reply:
Hmm, fair point!